presto-main/src/main/java/com/facebook/presto/type/TypeJsonUtils.java

```diff
@@ -284,6 +287,9 @@ else if (javaType == long.class) {
             if (element instanceof SqlDecimal) {
                 type.writeLong(blockBuilder, ((SqlDecimal) element).getUnscaledValue().longValue());
             }
+            else if (FLOAT.equals(type)) {
+                type.writeLong(blockBuilder, floatToRawIntBits(((Number) element).floatValue()));
```

Comment:
It was caught by the current Hive tests, but I'll look to see if there are any more specific unit tests we can add.